pkg/apis/app/v1beta1/application_types.go

```diff
@@ -84,15 +84,36 @@ type ApplicationStatus struct {
 
 // ImageSpec contains information about an image used as an icon.
 type ImageSpec struct {
-	// The source for image represented as either an absolute URL to the image or a Data URL containing
+	// Binary contains the base64-encoded image binary.
+	//
+	// Only one of Binary, Text, URL, or ConfigMapKeyRef may be defined.
+	Binary string `json:"binary,omitempty"`
+
+	// Text contains the image definition as plain text, for formats such as SVG.
+	//
+	// Only one of Binary, Text, URL, or ConfigMapKeyRef may be defined.
+	Text string `json:"text,omitempty"`
```

Comment:
The current spec supports Data URLs ([IETF RFC 2397](https://tools.ietf.org/html/rfc2397)). These support base64 encoded images and SVG as text. CSS-Tricks has an article titled, ["SVG with Data URIs"](https://css-tricks.com/lodge/svg/09-svg-data-uris/) showing how these work.

Data URLs are, also, something that can be passed to browsers as they know how to handle them.

The Go URL knows how to parse Dara URLs. See example at https://play.golang.org/p/fAyOHuMNaxW

How will this change improve upon the Data URLs already in the codebase?

Reply:
+1 Data URIs can be directly passed as a src to image tags, so I feel the existing field supports this